graphql/graphql_test.go

```diff
@@ -45,29 +44,116 @@ func TestBuildSchema(t *testing.T) {
 }
 
 // Tests that a graphQL request is successfully handled when graphql is enabled on the specified endpoint
-func TestGraphQLHTTPOnSamePort_GQLRequest_Successful(t *testing.T) {
+func TestGraphQLBlockSerialization(t *testing.T) {
 	stack := createNode(t, true)
 	defer stack.Close()
 	// start node
 	if err := stack.Start(); err != nil {
 		t.Fatalf("could not start node: %v", err)
 	}
-	// create http request
-	body := strings.NewReader("{\"query\": \"{block{number}}\",\"variables\": null}")
-	gqlReq, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s/graphql", "127.0.0.1:9393"), body)
-	if err != nil {
-		t.Error("could not issue new http request ", err)
-	}
-	gqlReq.Header.Set("Content-Type", "application/json")
-	// read from response
-	resp := doHTTPRequest(t, gqlReq)
-	bodyBytes, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		t.Fatalf("could not read from response body: %v", err)
+
+	for i, tt := range []struct {
+		name string
+		body string
+		want string
+		code int
+	}{
+		{
+			name: "get_number",
+			body: `{"query": "{block{number}}","variables": null}`,
+			want: `{"data":{"block":{"number":0}}}`,
+			code: 200,
+		},
+		{
+			name: "get_numeric_fields",
+			body: `{"query": "{block{number,gasUsed,gasLimit}}","variables": null}`,
+			want: `{"data":{"block":{"number":0,"gasUsed":0,"gasLimit":11500000}}}`,
+			code: 200,
+		},
+		{
+			name: "number_0",
+			body: `{"query": "{block(number:0){number,gasUsed,gasLimit}}","variables": null}`,
+			want: `{"data":{"block":{"number":0,"gasUsed":0,"gasLimit":11500000}}}`,
+			code: 200,
+		},
+		{
+			name: "number_-1",
+			body: `{"query": "{block(number:-1){number,gasUsed,gasLimit}}","variables": null}`,
+			want: `{"data":{"block":null}}`,
+			code: 200,
+		},
+		{
+			name: "number_-500",
+			body: `{"query": "{block(number:-500){number,gasUsed,gasLimit}}","variables": null}`,
+			want: `{"data":{"block":null}}`,
+			code: 200,
+		},
+		{
+			name: "string_0",
+			body: `{"query": "{block(number:\"0\"){number,gasUsed,gasLimit}}","variables": null}`,
+			want: `{"data":{"block":{"number":0,"gasUsed":0,"gasLimit":11500000}}}`,
+			code: 200,
+		},
+		{
+			name: "string_-33",
+			body: `{"query": "{block(number:\"-33\"){number,gasUsed,gasLimit}}","variables": null}`,
+			want: `{"data":{"block":null}}`,
+			code: 200,
+		},
+		{
+			name: "string_1337",
+			body: `{"query": "{block(number:\"1337\"){number,gasUsed,gasLimit}}","variables": null}`,
+			want: `{"data":{"block":null}}`,
+			code: 200,
+		},
+		// remove to allow hex string support
+		{
+			name: "string_1337",
+			body: `{"query": "{block(number:\"0xbad\"){number,gasUsed,gasLimit}}","variables": null}`,
+			want: `{"errors":[{"message":"strconv.ParseInt: parsing \"0xbad\": invalid syntax"}],"data":{}}`,
+			code: 400,
+		},
+		// uncomment to test hex string support
+		//{
+		//	name: "string_0x0",
```

Comment:
Yeah let's either uncomment or remove these. The tests run blazing fast now that they reuse the same stack, so IMO just uncomment them (and add whatever the response is) for more coverage.

Reply:
The code enabling this behavior is commented out too.

Reply:
My view on this: We should get the EIP updated before this gets uncommented.

Reply:
I'll push a change to show what I mean. If we don't support it, the tests should be there anyway, to 'cement' the behaviour of not supporting it. If we decide to support it, then we change the test. Tests are good to prevent accidental changes.